### DISCUSSION TITLE
„Full Solar‐Passthrough-Mode“ Wer hat Erfahrungen damit?

### DISCUSSION THREAD

**Discussion by SW-Niko** [hoylabs/OpenDTU-OnBattery#1149] [Q&A]
Hallo Community,
ich habe in den letzten Tagen eine Art „Full Solar‐Passthrough-Mode“ für den Absorption/Float Betriebszustand gemacht. Ich nenne ihn zu Unterscheidung „Surplus-Power-Mode".
Und wenn ich das Ergebnis meiner Bemühungen so betrachte, dann kommen mir Zweifel ob sich der Aufwand gelohnt hat. Weil der „Full Solar‐Passthrough-Mode“ und der „Surplus-Power-Mode" sich im Resultat zu ähnlich sind.

Der "Surplus-Power-Mode" regelt die Inverter-Leistung auf Basis der überschüssigen Solar-Leistung. Überschüssige Solar-Leistung ist immer dann vorhanden, wenn die Batterie (fast) voll ist und die verfügbare Solar-Leistung höher ist als die Leistung, die im Haushalt verbraucht wird.
Das Prinzip ist ganz einfach.
Ich hole mir von den MPPT den aktuellen Batterie-Lade-Mode. Und wenn das Absorption oder Float ist lese ich die im MPPT hinterlegte Grenz-Spannungen ein und regle den Inverter so das diese Spannung gerade nicht unterschritten wird.

Beispiel:
![grafik](https://github.com/user-attachments/assets/f394edfa-a74d-402c-820b-7a3720ec38ab)
Die (rote) Line ist die Inverter Leistung, die sich im Absorption/Float Mode aus dem höheren der beiden Werte Verbrauch (violett) oder Solar-Leistung (grün) ergibt. 

Und jetzt kommt meine Frage:
Was macht der „Full Solar‐Passthrough-Mode“ wenn die MPPT Leistung größer ist als die Inverter Leistung und der MPPT in den Absorption oder Float Mode kommt? Kann da jemand seine Erfahrungen mit mir teilen?

**Comment by Manos1966** [comment]
> Was macht der „Full Solar‐Passthrough-Mode“ wenn die MPPT Leistung größer ist als die Inverter Leistung und der MPPT in den Absorption oder Float Mode kommt? Kann da jemand seine Erfahrungen mit mir teilen?

Hi @SW-Niko ich habe so ein System laufen (Victron MPPT 250/60 und HM1500)

Ich schreibe ein paar Sachen (und @schlimmchen und die anderen werden mich korrigieren, falls ich "Bloedsinn" erzaehle... 🙈 )

"und der MPPT in den Absorption oder Float Mode kommt" spielt keine Rolle, OpenDTU-OnBattery sieht wieviel Watt der MPTT produziert und stellt den Wechselrichter auf das gleiche (minus Verluste Anpassungen usw).

Ist die MPPT Watt Leistung grosser als die maximale Leistung des Wechselrichters, dann
MPPT Leistung 2000W - Wechselrichter max 1500W =  500W gehen weiter an die Batterie, dadurch wird die Batterie weiter geladen.
Der Victron MPPT merkt das und faengt an, seiner Leistung anzupassen (Absorption und spaeter Float Mode).
Dadurch verringert sich die Victron MPPT Watt Leistung.
Ist die MPPT Watt Leistung weiterhin grosser als die maximale Leistung des Wechselrichters dann
MPPT Leistung 1600W - Wechselrichter max 1500W =  100W gehen weiter an die Batterie, dadurch wird die Batterie weiter geladen.
Der Victron MPPT merkt das und faengt an, seiner Leistung anzupassen (Float Mode), verringert sich die Victron MPPT Watt Leistung bis auf minimum.

**Im Prinzip, es ist das Standard Verhalten des Victron MPPT wenn die Batterie voll 100% wird (das Laden der Batterie hoert auf).**

Der Nachteil der " MPPT Leistung größer ist als die Inverter Leistung" ist also:
Man kann nicht die OpenDTU-OnBatterie nutzen um das Laden der Batterie bis auf einen selbst gewaehlten Wert (z.B. SOC 85% max) zu kontrollieren.

Notiz: Will man dies unbedingt (z.B. SOC 85% max und nicht mehr), in diesem Fall kann man den Victron MPPT steuern und die maximale Spannung Volt auf die Volt Spannung der Batterie bei 85% stellen. 😉 
Nachteil: Damit wird die Batterie langsamer geladen, weil der Victron haelt sich an diese niedrigere Spannung (anstatt der hoeheren Spannung die die Batterie aushalten koennte).

**Reply by SW-Niko** [reply]
Hallo @Manos1966 ,
danke für die Info. Das deckt sich mit meinen Überlegungen. Mir war nur nicht klar in welche Richtung das läuft.
Ich versuche mal meine Gedanken zusammenzufassen.

1. Im Bulk Mode verhält sich der Batterie Ausgang des MPPT wie eine Leistungs/Strom-Quelle. Wir bekommen vom MPPT die Info welche Leistung/Strom entnommen werden **kann.**
2. Absorption/Float mode. In diesen beiden Zuständen verhält sich der Batterie Ausgang des MPPT wie eine konstant Spannungsquelle mit Leistungs/Strom - Begrenzung. Wir bekommen vom MPPT die Info welche Leistung/Strom entnommen **wird.**

Es kommt darauf an wie genau wir die Verluste bestimmen und in die Leistungs-Bilanz einrechnen. 
MPPT-Leistung - Verluste = Inverter-Leistung
Es ergeben sich 3 Möglichkeiten.

1. Die berechneten Verluste stimmen ganz genau. Resultat: Die MPPT Leistung bleibt immer gleich auch wenn mehr Solar Leistung zur Verfügung stehen würde.
2. Die berechneten Verluste sind zu klein. Resultat: Die MPPT Leistung steigt immer weiter an bis wir an die aktuelle maximale Solar Leistung anstoßen.
3. Die berechneten Verluste sind zu groß. Die MPPT Leistung fällt immer weiter ab bis wir den Inverter auf die untere Leistungsgrenze heruntergefahren haben. (Der Fall der auf deinem System auftritt?)

Schwer verdaulich? Ich bin mir auch nicht sicher ob meine Überlegungen richtig sind.
Ich muss das bei schönen Wetter mal testen und die Berechnung der Verluste bewusst auf beide Seiten manipulieren.

>Der Victron MPPT merkt das und faengt an, seiner Leistung anzupassen (Float Mode), verringert sich die Victron MPPT Watt Leistung bis auf minimum.

Sehr guter Hinweis. Beim Übergang von Absorption in Float verringert der MPPT die Spannung. Das muss ich auch noch mal überdenken.

**Comment by Manos1966** [comment]
Ich verstehe nicht ganz, warum du deiner Zeit damit verbringst...

Die einzige Aufgabe des "Full Solar‐Passthrough-Mode" ist den Ladestand einer Batterie bis zur einer maximalen Prozent %, der kleiner als 100% ist, zu halten. Und dadurch die Batterie langfristig zu "schönen"...
Und das funktioniert sehr gut, so lange der Victron geringere Leistung liefert als der angeschlossene Wechselrichter.

In meiner Test-Anlage, der Victron 250/60 (3440W max Leistung) erreicht locker 2000W und das ist weit über die Max Leistung des angeschlossenen HM-1500.
![Victron MPPT 250-60](https://github.com/user-attachments/assets/5627c857-0109-43b5-a0f9-b5ecfc440768)

Die Verluste % zu berechnen und dementsprechend anzupassen ist leicht:
- Der Victron MPPT zeigt unter "Ausgang (Batterie)" wieviel er Richtung Wechselrichter/Batterie sendet.
- Der Wechselrichter zeigt unter DC-Leistung, oder durch die Strom Summe aller Strings wieviel rein kommt. 
- Die Batterie zeigt unter Strom, wieviel Strom rein in die Batterie (plus) oder raus aus der Batterie (minus) geht. 

Damit kann man die Prozentual Anpassung so anpassen, dass eine null oder so nah wie moeglich an +/- null, erreicht wird: "Victron MPPT Ausgang (Batterie)" = "Hoymiles DC Eingang", oder Batterie Strom = null

![Pylon Strom In-OUT](https://github.com/user-attachments/assets/3f32a8e6-34bf-4864-bb89-917b5fe4a853)

**Reply by SW-Niko** [reply]
>Die einzige Aufgabe des "Full Solar‐Passthrough-Mode" ist den Ladestand einer Batterie bis zur einer maximalen Prozent %, der kleiner als 100% ist, zu halten. Und dadurch die Batterie langfristig zu "schönen"...

Das stelle ich ja nicht in Frage. Das funktioniert.

Ich habe ein etwas anders Ziel:

- Inverter Leistung kleiner als MPPT Leistung
- Bei (fast) voller Batterie alles was der MPPT liefern kann in Batterie + Inverter pumpen
- Solle der Bedarf  (Haushalt) höher sein, als das was der MPPT liefern kann (Wolken), dann wird die fehlende Energie von der Batterie genommen

Und jetzt frage ich mich, ob ich mit dem "Full Solar‐Passthrough-Mode" meine Ziele erreichen kann. Ich habe gehofft das jemand aufspringt und sagt. "Ja das geht prima mit dem Full Solar‐Passthrough-Mode".

Das scheint bei dir ja prima zu klappen. Vielen Dank für die Info. 👍 
Das werde ich beim nächsten schönen Sonnentag antesten.

**Reply by Manos1966** [reply]
> Ich habe ein etwas anders Ziel:

Aaaaa! Warum hast du es nicht früher gesagt? 😄 

Du beschreibst die Funktion "Solar Passthrough"
(Bis jetzt hast du Fragen gestellt bezüglich der Funktion "Full Solar Passthrough")

`Wichtig vorab: DEAKTIVIERE die Option 'Use battery at night even if only partially charged' `
Der Rest ist hier beschrieben:
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Dynamic-Power-Limiter-%E2%80%90-Solar%E2%80%90Passthrough

Wichtige Definition vorab:
> Battery **charge cycle**: A battery charge cycle is started when the battery SoC or voltage falls below the respective stop threshold. The charge cycle completes when the battery SoC or voltage reaches the respective start threshold. <br>
Battery **discharge cycle**: The battery is or was charged to or beyond the start threshold. The discharge cycle ends when the battery SoC or voltage reaches the respective stop threshold.

> The inverter is **disabled** during the battery CHARGE cycle
The inverter is **enabled** during a battery DISCHARGE cycle

Ob du Solar Pass Through eingeschaltet hast, oder nicht: Der Wechselrichter wird aktiv erst wenn der DISCHARGE Zyklus gestartet ist.
(kleine Notiz am Rande: CHARGE / DISCHARGE Zyklus hier hat nichts mit dem Victron MPPT zu tun)


Du hast zwei Möglichkeiten zur Auswahl:
1. "Solar Passtrhough" deaktivieren. Dann, 
"Start Threshold for Battery Discharging" niedrig stellen (z.B. 40%), oder wenn du "Bei (fast) voller Batterie alles was der MPPT liefern kann in Batterie + Inverter pumpen" stelle das Wert, dass du als "(fast) volle Batterie" haben willst.
Nachteil: "Full Solar Passthrough" ist nicht als Option angeboten (deine Batterie wird/kann bis zu 100% geladen - ausser wenn du die Spannung im Victron MPPT niedriger stellst, was eine schlechtere Aufladeleistung bedeutet)

2. Du willst umbedingt "Full Solar Passthrough" aktiviert haben = Deine Batterie schoenen und unter SOC 100% aufladen, ohne die Ladeleistung des Victron MPPT dafuer einzuschraenken zu muessen? Dann,
  2.1 Solar Passthrough aktivieren
  2.2 Option 'Use battery at night even if only partially charged' deaktivieren (zur Zeit)
  2.3 "Start Threshold for Battery Discharging" niedrig stellen(z.B. 40%), oder wenn du "Bei (fast) voller Batterie alles was der MPPT liefern kann in Batterie + Inverter pumpen" haben willst, stelle den Wert, dass du als "(fast) volle Batterie" verstehst.

Der Unterschied zwischen Open 1 und 2
Bei Option 1: 
The inverter is **disabled** during the battery CHARGE cycle = Passiert gar nichts
The inverter is **enabled** during a battery DISCHARGE cycle
The inverter power limit is NOT constrained to the solar output, discharging of the battery happens<br>
(Solle der Bedarf (Haushalt) höher sein, als das was der MPPT liefern kann, dann wird die fehlende Energie von der Batterie genommen)

Bei Option 2: 
The inverter is **enabled** during the battery CHARGE cycle **so lange der Victron MPPT Strom liefert**
During CHARGE cycle, the inverter power limit is constrained to the solar output, such that discharging of the battery is avoided <br>
(Waehernd des CHARGE Zyklus, solle der Bedarf (Haushalt) höher sein, als das was der MPPT liefern kann, dann wird die fehlende Energie NICHT von der Batterie genommen)
The inverter is **enabled** during a battery DISCHARGE cycle. Jetzt willst du wissen ob waehrend des DISCHARGE Zyklus und solange es nicht NACHT ist, der Wechselrichter Strom auch von der Batterie entnehmen wird. 
**Antwort: Ja das macht er**
Solar Passthrough ist ON
Batterie SOC 70% 
Batterienutzung Start-Schwellwert 45 %
Batterienutzung Stop-Schwellwert 12 %
Full-Solar-Passthrough Start-Schwellwert  85 %
Also Tagsueber und DISCHARGE Zyklus
Der Victron MPPT liefert 376W, der HM-1500 verbraucht 1480,5W = nimmt den Unterschied aus der Batterie
![Test Solar Passthrough](https://github.com/user-attachments/assets/a365e0cd-12f6-4d67-8f7e-ae49c0b905da)

und nochmals zur Sicherheit mit Full Solar Passthrough aktiviert
Solar Passthrough ist ON
Batterie SOC 71% 
Batterienutzung Start-Schwellwert 45 %
Batterienutzung Stop-Schwellwert 12 %
Full-Solar-Passthrough Start-Schwellwert  50 %
Also Tagsueber und DISCHARGE Zyklus (und Full Solar Passthrough aktiv)
Der Victron MPPT liefert 599W, der HM-1500 verbraucht 1471,9W = nimmt den Unterschied aus der Batterie
![Test Solar Passthrough mit Full Solar Passthrough](https://github.com/user-attachments/assets/67151c8f-8675-44b8-9845-c0226a4aa809)


PS. Alles unter Vorbehalt bis @schlimmchen sein "Stamp of Approval" gegeben hat 😉

**Reply by SW-Niko** [reply]
@Manos1966 ,
vielen lieben Dank für die umfangreiche Hilfestellung. Ich teste das alles und berichte dann.
Wird aber etwas dauern. Bei meiner kleinen Anlage dauert es bis die Batterie voll ist und dann muss ich auch noch vor Ort zum testen sein.

**Reply by SW-Niko** [reply]
Hallo @Manos1966 ,
Ich denke, jetzt habe ich es verstanden und ich kann dir die Stellen im Wiki zeigen die mich aufs Glatteis geführt haben.

![grafik](https://github.com/user-attachments/assets/d5fd49cd-8b86-424b-82a7-6c3107ffbbd1)
Das passt nicht zu der dann folgenden Beschreibung. Ich würde es besser verstehen, wenn wir den Text ändern.
„As with Solar-Passthrough OFF, in general we have the battery CHARGE cycle, and the battery DISCHARGE cycle.”

![grafik](https://github.com/user-attachments/assets/b1935ad5-a1bd-49dc-8e1e-7127296dde2e)
Ich schlage vor folgendes zu ändern
As with Solar-Passthrough ON, in general we have nearly the same behavior.
The only difference happens during the battery DISCHARGE cycle when the battery reached or is above the Full Solar-Passthrough threshold:“

“The inverter's limit is set to the higher power, the total solar power output or the household consumption.

This means surplus solar power is fed into the grid.

Dann passt das auch zu deinen Messungen. Stimmst du mir zu?

**Reply by Manos1966** [reply]
Hallo @SW-Niko 
Willkommen im Club 🤣 🤣 🤣 🤣 
Wir haben uns alle die Zahne ausgebissen beim Versuch eine "Einfache und Verständliche" Beschreibung der Solar-Passthrough (mit/ohne "Full"), zu schreiben. 

Deine Vorschlaege sind gut, lassen aber Potenzial für Verwirrung an anderen Stellen:
„As with Solar-Passthrough OFF, in general we have the battery CHARGE cycle, and the battery DISCHARGE cycle.” erwaehnt nicht den Wechselrichter (wir kriegen mit Muehe und Not ueberhaupt die Leute dazu, das Prinzip CHARGE-, DISCHARGE-Zyklus zu lesen).

“(During the CHARGE cycle) The inverter's limit is set to the higher power, the total solar power output **or the household consumption**." Stimmt nicht, letzteres wird passieren erst wenn der "Batterienutzung Start-Schwellwert" erreicht ist (ich habe zum Beispiel manchmal "Batterienutzung Start-Schwellwert = 80%" und "Full-Solar-Passthrough Start-Schwellwert=60%").

Gibt dir keine Mühe! 🤣 
Nur diejenigen, die sich ausgiebig mit dem Thema befassen und testen (wie du, in diesem Fall) werden die Komplexität verstehen. Immerhin, es steckt eine Wahnsinnsidee (und Wahnsinnsprogrammierung) hinter der Solar Passthrough/Full Solar Passthrough und "Batterie nachts sogar teilweise geladen nutzen" Umsetzung!

**Comment by SW-Niko** [comment]
Hallo @Manos1966,
ich stimme dir ja zu ... wir können das niemals so beschreiben, das jeder es versteht ohne Fragen zu stellen.
Aber ich denke, wir sollten zu mindestens versuchen nichts falsches im Wiki zu haben.

PS: Über die Begriffe "CHARGE cycle" und "DISCHARGE cycle” bin ich anfangs auch gestolpert und ohne die ausführliche Beschreibung hätte ich die beiden Begriffe falsch interpretiert. Ich habe den Text vor einiger Zeit in eine Grafik umgesetzt.

![grafik](https://github.com/user-attachments/assets/2c8b927f-e382-41c7-841d-3be5d78ab0e5)


![grafik](https://github.com/user-attachments/assets/91be4a73-340c-4d58-82ba-3a8f99b62a7f)

Auf der X-Achse ist die Batterieladung in SoC und auf der Y-Achse die Zeit.
Wir starten mit einem Reboot bei einer SoC, die unter dem „Start Threshold“ liegt.
Die Farben der Pfeile zeigen den jeweiligen Zyklus.

Ist das Verständlich und könnte das Helfen die Begriffe leichter zu verstehen?

**Reply by Manos1966** [reply]
Guten Morgen @SW-Niko 

Danke fuer deine Muehe!
Leider nicht, nimmt die unterste Linie zum Beispiel, Charge Zyklus (Rot) ist korrekt dargestellt bis zum Start Threshold.
Danach faengt der Dicharge Zyklus (Gruen) an. Warum erhoeht sich der Batterie SOC% weiter (waehrend der Inverter ON ist)?
Dies **koennte** der Fall sein wenn der MPPT mehr liefert als der Inverter verbraucht, **muss aber nicht** der Fall sein...

**Reply by Manos1966** [reply]
Aber bleiben wir auf das wichtigste: Ist damit dein Ziel erreicht?

> Ich habe ein etwas anders Ziel:
Inverter Leistung kleiner als MPPT Leistung
Bei (fast) voller Batterie alles was der MPPT liefern kann in Batterie + Inverter pumpen
Solle der Bedarf (Haushalt) höher sein, als das was der MPPT liefern kann (Wolken), dann wird die fehlende Energie von der Batterie genommen
Und jetzt frage ich mich, ob ich mit dem "Full Solar‐Passthrough-Mode" meine Ziele erreichen kann. Ich habe gehofft das jemand aufspringt und sagt. "Ja das geht prima mit dem Full Solar‐Passthrough-Mode".

**Reply by spcqike** [reply]
ich finde die Darstellung sehr gut und gelungen. (auch wenn ich denke, dass die Zeit auf der X-Achse sinnvoller wäre. zumindest kenne ich es nur so aus diversen Datenblättern :D Spannung oder Strom auf Y, Zeit auf X)

@Manos1966 
Wie solle man denn den individuellen Bedarf eines Jeden darstellen? 

Vielleicht macht es Sinn und wird klarer, wenn man zusätzlich zum SoC (oder zur Spannung) auch die Leistung beispielhaft darstellen würde. ähnlich solcher FUnktionsdiagramme

![grafik](https://github.com/user-attachments/assets/4b3a7b29-ed20-4f93-8d04-89b95ef55a0c)

oben halt der Batterie(lade)zustand und unten drunter evtl. der beispielhafter Verbrauch am Haus und die Ent-/Ladeleistung aus/in der/die Batterie. Solange die Batterie lädt ist der Hausverbrauch halt .... vorhanden. Die Batterie hat einen (positiven?) Ladestrom. Wenn der Entladezyklus beginnt, kann die Batterie den Hausverbrauch decken, dieser geht auf 0, die Batterie hat einen (negativen?) Entladestrom. und so weiter.

**Reply by Manos1966** [reply]
> ich finde die Darstellung sehr gut und gelungen.

"We agree to disagree" @spcqike , wie kannst du sie gelungen finden, wenn sie eine Discharge Linie zeigt, die nach dem "Start Threshold" weitere Erhöhung des SOC% zeigt?

> Wie solle man denn den individuellen Bedarf eines Jeden darstellen?

Kann man nicht, wir haben viel zu unterschiedliche Ebenen an Anwendern bei diesem Projekt. Das grosste Teil wird abgedeckt in Diskussionen wie diese hier. **Voraussetzung** dafuer ist das der Anwender seine "Hausarbeit" bereits gemacht hat (wie SW_Niko hier gemacht hatte, als er die Diskussion angefangen hat).
Wie ich bereits erwaehnt habe, "Nur diejenigen, die sich ausgiebig mit dem Thema befassen und testen werden die Komplexität verstehen. Immerhin, es steckt eine Wahnsinnsidee (und Wahnsinnsprogrammierung) hinter der Umsetzung Solar Passthrough/Full Solar Passthrough und Batterie nachts sogar teilweise geladen nutzen"

**Reply by spcqike** [reply]
> wie kannst du sie gelungen finden, wenn sie eine Discharge Linie zeigt, die nach dem "Start Threshold" weitere Erhöhung des SOC% zeigt?

ich habe 6000Wp und 5W Grundlast. Warum sollte der SoC nicht steigen? ;)

**Reply by SW-Niko** [reply]
Sorry Leute, ich wollte hier kein Fass aufmachen und euch die Zeit rauben. 😞 

@Manos1966 
>Dies koennte der Fall sein wenn der MPPT mehr liefert als der Inverter verbraucht, muss aber nicht der Fall sein...

Guter Einwand. Diese Info fehlt. Die Grafik zeigt nur konstante Energie in die Batterie, konstante Energie aus der Batterie und keine Energie in die Batterie. Der Inverter wird nicht berücksichtigt.

>Aber bleiben wir auf das wichtigste: Ist damit dein Ziel erreicht?

Ja, das habe ich. Ich kenne jetzt die Unterschiede zwischen dem „Full Solar‐Passthrough-Mode“ und meinem „Surplus-Power-Mode" und ich denke ich werde den „Surplus-Power-Mode" in einem PR fassen.

@spcqike 
>ich finde die Darstellung sehr gut und gelungen. (auch wenn ich denke, dass die Zeit auf der X-Achse sinnvoller wäre. zumindest kenne ich es nur so aus diversen Datenblättern :D Spannung oder Strom auf Y, Zeit auf X)

Zeit auf der X-Achse geht natürlich auch. Ich habe das in erster Linie mal für mein Verständnis gemacht um die Unterschiede zu verstehen.
Beim "Full Solar-Passthrought" bin ich dann hier gelandet.

![grafik](https://github.com/user-attachments/assets/59aab2b9-6f46-458b-a356-1b7a4722ee4e)

Und letztendlich gescheitert bei der Frage was passiert wenn man „Full Solar-Passthrough Start Threshold“ unter dem „Start Threshold for Battery Discharging“ konfiguriert. 😁

**Reply by Manos1966** [reply]
@SW-Niko Du versuchts, ein Multidimensionales System auf eine Zweidimensionale Ebene darzustellen.

**Reply by Manos1966** [reply]
> ich habe 6000Wp und 5W Grundlast. Warum sollte der SoC nicht steigen? ;)

Ich habe keine Zeit fuer Sarkasmus, ich bin raus.

**Reply by spcqike** [reply]
> Ich habe keine Zeit fuer Sarkasmus, ich bin raus.

tut mir leid, wenn du meinen, zugegeben leicht überspitzten, Kommentar negativ aufnimmst. aber auch dein  ``wie kannst du sie gelungen finden, wenn sie eine Discharge Linie zeigt, die nach dem "Start Threshold" weitere Erhöhung des SOC% zeigt?`` ist nicht zielführend. 

wie du selbst richtig anmerkst, ist die Funktion und Regelung sehr vielschichtig und mit Nichten nur schwarz/weiß. Ob die Batterie weiter geladen wird hängt, wie du sicher besser weißt als ich, von unzähligen Faktoren ab, die in dieser Darstellung so gar nicht erfassbar sind. 

Wäre ein sofortiges Entladen deiner Meinung nach denn besser oder genauer? Ist das überhaupt realistisch?

Ich stelle die Vermutung in den Raum, dass die meisten Batterien zur Mittagszeit soweit geladen sind, dass der Startwert überschritten wird. Zu dieser Zeit liefert die PV recht viel Energie, der Verbrauch ist zumeist jedoch gering. also warum würde die Batterie entladen?

Selbst mit meiner alten 2x425Wp Anlage wäre die Erzeugung am frühen Vormittag (also nach dem Aufladen) weit über meiner Grundlast von recht hohen 350W. Bei aktuell 8x435Wp ist das nochmal n Zacken stärker. (nur leider fehlt mir noch immer der Akku 😄 )

Aber, wie gesagt, das wäre mein Szenario, das mag bei anderen Usern anders aussehen. 


> Du versuchts, ein Multidimensionales System auf eine Zweidimensionale Ebene darzustellen.

Das macht man doch immer, oder? (mit Außnahme weniger 3-dimensionaler Diagramme)
Wie ich oben meinte, verständlicher wird es bestimmt, wenn man auch die Lade- und Entladeleistungen und den Bedarf des Haushalts dazu darstellt. Bzw. noch die anliegende Solarleistung.

Eventuell stellt ja jemand mit einem entsprechenden System seine Daten aus Grafana / als Diagramm zur Verfügung. ungefähr so

![grafik](https://github.com/user-attachments/assets/ef508329-575d-492d-aa71-3557d7a4721a)

Solange die PV Leistung > Hausbedarf ist, muss die Batterie ja weiter geladen werden. wenn der Bedarf < PV Leistung, wird sie entladen. natürlich ist das alles sehr komplex und verwurren. Aber der ein oder andere User hat vielleicht ein entsprechendes Monitoring und kann solch Diagramme zusammenklicken.

**Reply by SW-Niko** [reply]
Ach Leute,
gerade wegen der unterschiedlichen Ansichten finde ich so eine Diskussion sehr interessant.

@Manos1966 
>Du versuchts, ein Multidimensionales System auf eine Zweidimensionale Ebene darzustellen.

Richtig. Ein multidimensionales System kann man nicht auf einer Zweidimensionale Ebene darzustellen. Da bin ich bei dir.
Das war aber nicht meine Absicht. Ich wollte nur diesen einen Aspekt "CHARGE cycle" und "DISCHARGE cycle” darstellen. 

@spcqike 
Dein Ansatz mit mehreren Grafiken ist nicht falsch aber vieleicht auch wieder zu komplex wenn man nur die beiden Begriffe "CHARGE cycle" und "DISCHARGE cycle” darstellen will.

Ich schlage vor, wir lassen das Thema erst mal ruhen, nehmen die verschiedenen Blickwinkel in uns auf und warten bis jemand eine bessere Idee hat.

**Comment by SW-Niko** [comment]
Was denkt ihr über diese Variante?
Darin habe ich jetzt @spcqike  Vorschlag mit X-Achse = Zeit und @Manos1966 Einwand mit dem Inverter berücksichtigt.
Genauer ... klargestellt das hier nur die Batterie berücksichtigt wird.

![grafik](https://github.com/user-attachments/assets/b640927c-e28f-45f9-a960-ae52431f2c7d)
